library/core/src/cell/lazy.rs

```diff
@@ -56,6 +56,31 @@ impl<T, F: FnOnce() -> T> LazyCell<T, F> {
         LazyCell { cell: OnceCell::new(), init: Cell::new(Some(init)) }
     }
 
+    /// Consumes the lazy, returning the evaluated value.
+    ///
+    /// Returns `Err(init)` if the lazy was never evaluated.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(once_cell)]
+    ///
+    /// use std::cell::LazyCell;
+    ///
+    /// let hello = "Hello, World!".to_string();
+    ///
+    /// let lazy = LazyCell::new(|| hello.to_uppercase());
+    ///
+    /// assert_eq!(&*lazy, "HELLO, WORLD!");
+    /// assert_eq!(LazyCell::into_inner(lazy).ok(), Some("HELLO, WORLD!".to_string()));
+    /// ```
+    #[unstable(feature = "once_cell", issue = "74465")]
```

Comment:
Should this maybe have a separate feature gate? When we'll be stabilizing once cell we'll want to first stabilize some sort of minimal API and only then stabilize other methods...
```suggestion
    #[unstable(feature = "once_cell_into_inner", issue = "74465")]
```

Reply:
No, it goes the other way around: https://github.com/rust-lang/rust/pull/105587

Reply:
That's just because we didn't split originally though?.. It's always easier to stabilize small parts of the API and it's also easier when you don't have to change the gate before stabilizing.

Reply:
Sure, but why should this method be stabilized separately from the rest of Lazy*? That seems like a decision that can be made when we're actually ready to stabilize the Lazy variants.

Reply:
That's just generally a good idea to have small feature gates, stabilizing multiple at the same time is a lot simpler and more clear than splitting one into multiple and stabilizing a subset. This method doesn't look essential, so I don't think it is likely it will be included in the first stabilization.